### DISCUSSION TITLE
HA Not Reporting 'Last Seen' Accurately for Dead Devices - OK in ZUI

### DISCUSSION THREAD

**Discussion by pumrum** [zwave-js/zwave-js#6688] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

Z-Wave JS UI reports a device that has gone missing (dead battery or otherwise) accurately within the ZUI interface by displaying the 'Last Active' timestamp. This survives restarts, reboots, firmware updates, and all other activity. However, the corresponding 'Last Seen' entity is not properly refreshed in HA and may report it as state 'Unknown'.

Thread discussion of the issue on Discord here:
https://discord.com/channels/330944238910963714/1211823234614431815

### Software versions

Driver (node-zwave-js): ...

Z-Wave JS UI: ...

Home Assistant Z-Wave Integration: ...

Home Assistant Z-Wave JS Addon: ...

ioBroker.zwave2 Adapter: ...

If you are using something non-standard, tell us here: ...
nothing non-standard.

zwave-js-ui: 9.9.0
zwave-js: 12.4.4
home id: 3457030142
home hex: 0xce0e17fe
Core 2024.2.1
Supervisor 2024.02.0
Operating System 11.5
Frontend 20240207.1

### Device information

all devices - in the case documented below, Node 017 had a battery die on Feb 6, and it is reported accurately in ZUI, but in HA the 'Last Seen' is 'Unknown'.

System information attached:
[sysinfo.txt](https://github.com/zwave-js/node-zwave-js/files/14413029/sysinfo.txt)

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[config_entry-zwave_js-a12b99b19a22b4d9282339f174c39045.json](https://github.com/zwave-js/node-zwave-js/files/14412987/config_entry-zwave_js-a12b99b19a22b4d9282339f174c39045.json)

[node009_debug.txt](https://github.com/zwave-js/node-zwave-js/files/14413037/node009_debug.txt)

[a comment by an automated account is not shown]

**Comment by kpine** [comment]
ZUI gets the baseline "last active" value from the node property `lastSeen`. It then keeps it updated via the node statistics updated event.

HA gets the baseline "last active" from the node `statistics.lastSeen` property/value. It also keeps it updated via the node statistics updated event.

From what I can tell, the node `statistics` are not persisted to disk cache, but `lastSeen` is. That means that after a Z-Wave JS restart, all the statistics are lost and so is "last seen", however ZUI works because the node property is still available. Mains-powered devices will quickly be resolved when they are pinged on startup, but dead or battery devices will not. This means HA won't have a valid value until any node updates, defeating the purpose of the sensor.

So the question is: is the current behavior expected and so HA needs to be fixed (or zwave-js-server), or can the driver keep the lastSeen property and statistics.lastSeen property in sync, even after restarts.

**Reply by kpine** [reply]
ZUI has a similar problem for its MQTT topic. It only publishes the topic when the statistics event is emitted.

I suppose both apps should use the node property on startup.

**Reply by AlCalzone** [reply]
@robertsLando @raman325 Last seen should default to `node.lastSeen` if the statistics aren't populated

**Reply by pumrum** [reply]
I'm running HAOS 12.3 and 2024.5.4 - and last_seen is still getting lost on host reboot. as mentioned above, mains powered devices come back on startup ping, but battery devices revert back to 'unknown' until the next time they're seen. of note: just restarting HA (not a full host reboot) does not suffer this issue, so it seems that [PR911](https://github.com/home-assistant-libs/zwave-js-server-python/pull/911) solved some of the issue

**Reply by robertsLando** [reply]
I already set it to node.lastSeen value on startup: https://github.com/zwave-js/zwave-js-ui/blob/724045c8bbb1b36112893447e7f242cb50bf588a/api/lib/ZwaveClient.ts#L5958

I wonder if I also should preiodically update it? If so it may make sense if the driver outputs the statistics updated event every time the lastSeen changes?

**Reply by AlCalzone** [reply, marked as answer]
--> https://github.com/home-assistant/core/issues/119390

I think the Z-Wave JS UI behavior is correct. HA loses the state whenever the addon or Z-Wave JS Server is restarted (probably when the integration reloads).
